pkg/routeagent/controllers/route/driver.go

```diff
@@ -3,32 +3,57 @@ package route
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 
 	"github.com/vishvananda/netlink"
 	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/util"
 )
 
-const FlannelInterface = "flannel.1"
+func interfaceExists(iface string) bool {
+	_, err := netlink.LinkByName(iface)
+	return err == nil
+}
 
-func isFlannelCNI() error {
-	_, err := netlink.LinkByName(FlannelInterface)
-	if err != nil {
-		return fmt.Errorf("error retrieving link by name %s: %v", FlannelInterface, err)
+func discoverCNIInterface(clusterCIDR string) string {
+	cniInterfaceList := []string{
+		"weave",     // Weave
+		"tun0",      // OpenShift
+		"flannel.1", // Flannel
 	}
 
-	// FlannelInterface exists, return nil
-	return nil
-}
+	for _, iface := range cniInterfaceList {
+		if interfaceExists(iface) {
+			klog.Infof("Found CNI interface [%s] on the Host", iface)
+			ipv4Addr, err := util.GetIPv4AddressOnInterface(iface)
+			if err != nil {
+				klog.Errorf("error reading IPv4 address on CNI interface [%s]: %v", iface, err)
```

Comment:
```suggestion
				klog.Errorf("error reading IPv4 address on CNI interface %q: %v", iface, err)
```

nit - so use quoted string instead of []? Same in other places.